components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1036,6 +1040,8 @@ func prepareIDELaunch(cfg *Config, ideConfig *IDEConfig) *exec.Cmd {
 	cmd.SysProcAttr.Setpgid = true
 	cmd.SysProcAttr.Pdeathsig = syscall.SIGKILL
 
+	cmd.SysProcAttr.AmbientCaps = append(cmd.SysProcAttr.AmbientCaps, unix.CAP_SYS_PTRACE)
```

Comment:
:bubbles: Would it make sense to have a central function to append the CAP_SYS_PTRACE? That way, we could also add a comment on _why_ we add it, and don't have to repeat ourselves.

Reply:
> so I suspect whether this is worth it.

Sometimes a single call can warrant a function, e.g. for documentation purposes or improving referenceability.

> Only this and the ssh use case are the same

I thought about:
```
// comment about the WHY
func grantCapSysPtrace(caps []uintptr) []uintptr {
     return append(caps, unix.CAP_SYS_PTRACE)
]